src/test/java/de/tum/in/www1/artemis/authorization/AuthorizationTestService.java

```diff
@@ -67,7 +93,14 @@ public boolean validEndpointToTest(HandlerMethod method, boolean onlyConditional
      * @return true if the endpoint depends on a profile, false otherwise
      */
     private boolean isConditionalEndpoint(HandlerMethod handlerMethod) {
-        return handlerMethod.getMethod().getAnnotation(Profile.class) != null || handlerMethod.getMethod().getDeclaringClass().getAnnotation(Profile.class) != null;
+        var methodProfileAnnotation = handlerMethod.getMethod().getAnnotation(Profile.class);
+        var classProfileAnnotation = handlerMethod.getMethod().getDeclaringClass().getAnnotation(Profile.class);
+        // No null-check required for classes because we have tests ensuring Profile annotations on classes
+        return (methodProfileAnnotation != null && isNonCoreProfile(methodProfileAnnotation)) || isNonCoreProfile(classProfileAnnotation);
+    }
+
+    private boolean isNonCoreProfile(Profile profileAnnotation) {
+        return !(profileAnnotation.value().length == 1 && profileAnnotation.value()[0].equals(PROFILE_CORE));
     }
```

Comment:
Is this correct? If we have more than 1 profile, we are non-core?

Reply:
We don't have a "core only" anymore. I don't mind if some endpoints get tested multiple times. I just want to avoid that all core endpoints get tested five times.